### PR TITLE
Cleaned up inner class based on suspicous memory behavior in idle and…

### PR DESCRIPTION
**Pull request by dpwspoon**
… parent sessions

### CODE REVIEW

File: transport/spi/src/main/java/org/kaazing/gateway/transport/AbstractBridgeSession.java [hunk 1]
```diff
@@ -122,7 +122,8 @@ private AbstractBridgeSession(int ioLayer,
         
         this.service = service;
         this.processor = processor;
-        this.parent = new AtomicReference<>(parent);
+        final AtomicReference<IoSessionEx> parentReference = new AtomicReference<>(parent);
```
Comment (jitsni) [inline transport/spi/src/main/java/org/kaazing/gateway/transport/AbstractBridgeSession.java]: you don't require parentReference. Just pass in parent to BridgeSessionConfigEx

**Comment by dpwspoon** [comment]
@jitsni Fixed, thanks.  And added a few related fixes

File: transport/ssl/src/main/java/org/kaazing/gateway/transport/ssl/SslConnector.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ public void setSecurityContext(SecurityContext securityContext) {
             // avoid caching SSLSession in shared SSLContextFactory instance
             // Note: SSLSessionContext.setSessionCacheSize(0) means unlimited,
             // so we use 1 instead
-            sslContextFactory.setServerSessionCacheSize(1);
+            sslContextFactory.setClientSessionCacheSize(1);
```
Comment (jitsni) [inline transport/ssl/src/main/java/org/kaazing/gateway/transport/ssl/SslConnector.java]: Do you want to move the ssl change into a separate PR ?

Comment (dpwspoon) [inline transport/ssl/src/main/java/org/kaazing/gateway/transport/ssl/SslConnector.java]: sure, will do

Comment (dpwspoon) [inline transport/ssl/src/main/java/org/kaazing/gateway/transport/ssl/SslConnector.java]: It is now here: https://github.com/kaazing/gateway/pull/813.  I'll change this PR request with the other issues to remove the TLS fix

**Comment by dpwspoon** [comment]
Closing in favor of https://github.com/kaazing/gateway/pull/814
